devices/Ds18b20/Ds18b20.cs

```diff
@@ -0,0 +1,536 @@
+
+                if (verificationValue == 0)
+                {
+                    return false;
+                }
+
+                while (_oneWireHost.ReadByte() == 0)
+                {
+                    Thread.Sleep(10);
+                }
+            }
+
+            SelectDevice();
+            verificationValue = _oneWireHost.WriteByte((byte)FunctionCommands.ReadScratchpad);
+
+            if (verificationValue == 0)
+            {
+                return false;
+            }
+
+            // Discard temperature bytes
+            _oneWireHost.ReadByte();
+            _oneWireHost.ReadByte();
+
+            _tempHighAlarm = (sbyte)_oneWireHost.ReadByte();
+            TemperatureHighAlarm = Temperature.FromDegreesCelsius(_tempHighAlarm);
+            _tempLowAlarm = (sbyte)_oneWireHost.ReadByte();
+            TemperatureLowAlarm = Temperature.FromDegreesCelsius(_tempLowAlarm);
+            int configReg = _oneWireHost.ReadByte();
+
+            if (_oneWireHost.TouchReset())
+            {
+                TemperatureResolution = (TemperatureResolution)(configReg >> 5);
+            }
+            else
+            {
+                TemperatureResolution = TemperatureResolution.VeryHigh;
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Writes sensor configuration. The unchanged registers will be overwritten.
+        /// </summary>
+        /// <param name="writeToEEPROM">Flag indicating if configuration should be written also to EEPROM.</param>
+        public void ConfigurationWrite(bool writeToEEPROM = false)
+        {
+            SelectDevice();
+
+            _oneWireHost.WriteByte((byte)FunctionCommands.WriteScratchpad);
+
+            _oneWireHost.WriteByte((byte)_tempHighAlarm);
+            _oneWireHost.WriteByte((byte)_tempLowAlarm);
+            _oneWireHost.WriteByte((byte)((byte)TemperatureResolution << 5));
+
+            if (writeToEEPROM)
+            {
+                SelectDevice();
+                _oneWireHost.WriteByte((byte)FunctionCommands.CopyScratchpad);
+                Thread.Sleep(10);
+            }
+        }
+
+        /// <summary>
+        /// Starts tracking the changes of temperature.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">When tracking is already enabled.</exception>
+        /// <exception cref="ArgumentOutOfRangeException">When tracking interval is shorter than 50 milliseconds.</exception>
+        /// <exception cref="NotSupportedException">When event handler is not assigned.</exception>
+        /// <param name="trackingInterval">Interval to track the changes to sensor values.</param>
+        public void BeginTrackChanges(TimeSpan trackingInterval)
+        {
+            if (_isTrackingChanges)
+            {
+                throw new InvalidOperationException("Already tracking changes");
```

Comment:
We can get empty strings here and leverage intellisense to make some space. Same for all the other exceptions where it makes sense.

Reply:
I don't understand what do you mean by "we can get empty strings here", can you elaborate?

Reply:
@rkalwak unless there is a very good reason we shouldn't output string messages on the exceptions. Reason: it takes too much flash space for usually not much usage. Instead, add the exception to the Intellisense comment along with the explanation on why one would see it being thrown. (in this case add InvalidOperationException).